### DISCUSSION TITLE
Production builds and prerender-manifest.json with PUT requests

### DISCUSSION THREAD

**Discussion by cfeltz34** [vercel/next.js#65315] [App Router]
I think that it's a bug but I don't know how to create a NextJS project.

I define a route like this : 
File app\api\...\edit\route.ts
```
import { routes } from "@/helpers/routes";
import { NextRequest, NextResponse } from "next/server";

export async function GET(req: NextRequest, res: NextResponse): Promise<Response> {
  // Return static data
  return NextResponse.json({}); 
}

export async function PUT(req: NextRequest, res: NextResponse): Promise<Response> {
  const body = await req.json();
  ..............
}
```

When I build for production, a "prerender-manifest.json" file is generate with static routes
The routes array have "api\...\edit"

When I debug generated code, in node_modules\next\dist\server\base-server.js file : 
```
   function renderToResponseWithComponentsImpl(...) {
        ....
        isSSG ||= !!prerenderManifest.routes[pathname === "/index" ? "/" : pathname];
        ...
        if (// Server actions can use non-GET/HEAD methods.
        !isServerAction && // Resume can use non-GET/HEAD methods.
        !minimalPostponed && !is404Page && !is500Page && pathname !== "/_error" && req.method !== "HEAD" && req.method !== "GET" && (typeof components.Component === "string" || isSSG)) {
            res.statusCode = 405;
            res.setHeader("Allow", [
                "GET",
                "HEAD"
            ]);
            await this.renderError(null, req, res, pathname);
            return null;
        }
   }
```
isServerAction = false
minimalPostponed = undefined
is404Page = is500Page = false
req.method = PUT
isSSG = true (because prerenderManifest.routes contains "api\...\edit")

--> application return 405 Not Allow status

Remark : If in my app\api\...\edit\route.ts file, I replace my PUT function by a POST function (or if I add a POST function), the file prerender-manifest.json doesn't contains my route and there are no problem.

**Comment by icyJoseph** [comment]
Could you try fixing the function signatures first?

In App Router, there's no response as second argument.

Another question, what's the route again? `/api.../edit` like that? with triple .?

**Reply by cfeltz34** [reply]
I just deleted the second parameter and it's the same.
My route is "/api/user/account/edit"
